### PR TITLE
API test data: incomplete_references param sets all references as not_requested_yet

### PR DESCRIPTION
**Pull request by carlosmartinez**
## Context

We are changing references such that the vendor API will include incomplete references (in `not_requested_yet` status, with referee info but not reference detail). We want to trial this structure in the test data to allow vendors to test on Sandbox.

## Changes proposed in this pull request

Add new param `incomplete_references` to `POST /test-data/generate` endpoint. When `true`, set all references in the generated applications to `not_requested_yet` and mark them as `selected`.

Also add param `incomplete_references` to `GET /applications` and `GET /applications/:id` endpoints so that they can optionally include references for which feedback has not been provided.

## Guidance to review

Run `/api/v1/test-data/generate` locally or on the review app, with `not_requested_yet` param set to true. Does it generate applications in the expected state?

## Link to Trello card

https://trello.com/c/2XfUgQbb/412-create-test-data-for-vendors-to-check-that-references-changes-will-not-cause-issues

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/lib/test_applications.rb [hunk 1]
```diff
@@ -215,18 +218,22 @@ def create_application_to_courses(
...
-      @application_form.application_references.feedback_requested.each do |reference|
-        submit_reference!(reference.reload)
-        reference.update(selected: true)
-        fast_forward
+        # Cancel 1 reference manually and receive feedback on the remaining 2.
+        # Select the two references with feedback.
+        @application_form.application_references.feedback_requested.first.cancelled!
+        @application_form.application_references.feedback_requested.each do |reference|
+          submit_reference!(reference.reload)
+          reference.update(selected: true)
```
Comment (JR-G) [inline app/lib/test_applications.rb:232]: I've just been testing this on Postman. I can see it correctly generates the test data but when you GET an application_choice it only pulls through 'Selected' references'?

Comment (carlosmartinez) [reply app/lib/test_applications.rb:235]: thanks @JR-G. have rather hackily resolved this by setting incomplete references as `selected = true`. given it's just a temporary thing for testing purposes, i think probably is ok.

**Comment by tomas-stefano** [comment]
Cool tested locally.

Generate test-applications on 13:55:

```
curl -X POST http://localhost:3000/api/v1/test-data/generate?not_requested_yet=true -H "Authorization: Bearer myawesomebearertoken"
```

Which leads to the response:

```
{"data":{"message":"Request submitted. Applications will appear once they have been generated"}}
```

Then sidekiq kicks-in! And then requesting the applications created on the first step:

```
curl http://localhost:3000/api/v1/applications?since=2022-07-27T13:55:40  -H "Authorization: Bearer myawesomebearertoken" | jsonpp | grep '"references"' -n20
```

And they all appear:

```
... Many references....
...
...
"references": [
31468-          {
31469-            "id": 105843,
31470-            "name": "Lyle VonRueden",
31471-            "email": "e3c872bef0@example.com",
31472-            "relationship": "Fugit dolorem nihil. Amet fugit qui. Tempore ut quasi.",
31473-            "reference": "You are awesome",
31474-            "referee_type": "professional",
31475-            "safeguarding_concerns": false
31476-          },
31477-          {
31478-            "id": 105846,
31479-            "name": "Antwan Feeney",
31480-            "email": "a94542023d@example.com",
31481-            "relationship": "Autem expedita dicta. Itaque iure eius. Modi libero maiores.",
31482-            "reference": "You are awesome",
31483-            "referee_type": "academic",
31484-            "safeguarding_concerns": false
31485-          }
31486-        ]
```

@carlosmartinez like the reference property above, you are awesome! 🚀  cc @JR-G

**Comment by tomas-stefano** [comment]
Tested with incomplete references now:

```
curl -X POST http://localhost:3000/api/v1/test-data/generate?incomplete_references=true -H "Authorization: Bearer XcfMdC12vsX6CcnPBGzC"
```

All good the reference is null:

```
"references": [
3437-          {
3438-            "id": 106824,
3439-            "name": "Ezequiel Kub",
3440-            "email": "ef668b6b27@example.com",
3441-            "relationship": "Autem ut nihil. Nostrum quod sapiente. Incidunt sed illo.",
3442-            "reference": null,
3443-            "referee_type": "academic",
3444-            "safeguarding_concerns": false
3445-          },
3446-          {
3447-            "id": 106826,
3448-            "name": "Leena Torphy",
3449-            "email": "1dac5ab858@example.com",
3450-            "relationship": "Enim aliquid dicta. Quasi ipsa corporis. Ducimus amet quidem.",
3451-            "reference": null,
3452-            "referee_type": "character",
3453-            "safeguarding_concerns": false
3454-          },
```

And verifying the references above into the DB:

```
ApplicationReference.find(106826).feedback_status
  ApplicationReference Load (15.7ms)  SELECT "references".* FROM "references" WHERE "references"."id" = $1 LIMIT $2  [["id", 106826], ["LIMIT", 1]]
=> "not_requested_yet"
```

So all good. cc @JR-G @carlosmartinez

**Review by tomas-stefano — APPROVED** [review]

**Comment by tomas-stefano** [comment]
🚀 🚀 🚀 🚀 🚀

**Pull request merged**